### PR TITLE
Fix compatibility scores.

### LINKED ISSUE #438: Sort Guides by Compatibility
Woo!!! Compatibility scores are working, now they should be weighted in the search results. This is probably an elasticsearch thang?
screenshot 2015-01-19 at 2 43 25 pm

### LINKED ISSUE #440: Basic Needs Compatibility is buggy
I have only one Garden, with 'Full Sun'. If I go to one guide with the 'Full Sun' Basic Need, it says I'm compatibile. If I go to another Guide also with the 'Full Sun' Basic Need, it says I'm not compatible because 'I have Partial Sun' according to the tooltip.

Hmmm?

### PR DESCRIPTION
**Pull request by guitsaru**
Compatibility scores were being applied to the owner of the guide rather
than to the user currently viewing the guides.

Because of this fix, we can also now sort by compatibility in the search results.

Fixes #440
Fixes #438

### CODE REVIEW

File: app/controllers/crop_searches_controller.rb [hunk 1]
```diff
@@ -12,15 +12,12 @@ def search
...
+                        )
     if query.empty?
       @crops = Crop.search('*', limit: 25, boost_by: [:guides_count])
     end
 
-    # Use the crop results to look-up guides
-    crop_ids = @crops.map { |crop| crop.id }
-    @guides = Guide.search('*',
-                           where: { crop_id: crop_ids })
+    @guides = GuideSearch.search('*').for_crops(@crops).with_user(current_user)
```
Comment (warpling) [inline app/controllers/crop_searches_controller.rb]: This is a great improvement :)

**Comment by simonv3** [comment]
Thanks for the PR! Looks good.

I just pulled in Rory's outstanding PR and it looks like it has some clashes with yours.

I'm going to take some time to properly absorb what's going on here. @RickCarlino maybe you have some thoughts?

File: app/models/garden.rb [hunk 2]
```diff
@@ -4,6 +4,8 @@ class Garden
   belongs_to :user
   validates_presence_of :user
 
+  after_save :reindex_guides
```
Comment (simonv3) [inline app/models/garden.rb]: At the moment it's probably okay to re-index all the guides every time a garden changes, but does it make sense to do so once our guides DB starts getting larger?

**Comment by simonv3** [comment]
I'm going to merge this in, but would love to hear your thoughts on my question above (reindexing on save). 

Thanks for all of this work! Once I pull it in, our staging server should deploy with it (http://staging.openfarm.cc)

File: app/models/garden.rb (hunk 2 above)
Comment (guitsaru) [inline app/models/garden.rb]: That's something I need to look into.  I know with resque or sidekiq you can set it up to only run one of these jobs at a time.  I'm not sure about delayed job.

Comment (simonv3) [inline app/models/garden.rb]: Cool, let us know if you think of anything, so that even if you don't have time someone else can try to figure it out :)

**Pull request merged**